ssl/record/methods/tls_common.c

```diff
@@ -916,11 +916,17 @@ int tls_get_more_records(OSSL_RECORD_LAYER *rl)
         }
 
         /*
-         * Check if the received packet overflows the current
-         * Max Fragment Length setting.
-         * Note: rl->max_frag_len > 0 and KTLS are mutually exclusive.
+         * Record overflow checking (e.g. checking if
+         * thisrr->length > SSL3_RT_MAX_PLAIN_LENGTH) is the responsibility of
+         * the post_process_record() function above. However we check here if
+         * the received packet overflows the current Max Fragment Length setting
+         * if there is one.
+         * Note: rl->max_frag_len != SSL3_RT_MAX_PLAIN_LENGTH and KTLS are
+         * mutually exclusive. Also note that with KTLS thisrr->length can
+         * be > SSL3_RT_MAX_PLAIN_LENGTH (and rl->max_frag_len must be ignored)
          */
-        if (thisrr->length > rl->max_frag_len) {
+        if (rl->max_frag_len != SSL3_RT_MAX_PLAIN_LENGTH
```

Comment:
Given the documentation language above, specifically "Also not that with KTLS thisrr->length can be > SSL3_RT_MAX_PLAIN_LENGTH (and rl->max_frag_len must be ignored).  Given that, shouldn't this check be reversed so that thisrr->length is checked first, and rl->max_frag_len is only interrogated if the former is > SSL3_RT_MAX_PLAIN_LENGTH?

Reply:
Not really sure why? It makes more sense to me this way. Checking `max_frag_len != SSL3_RT_MAX_PLAIN_LENGTH` is a test to see if max_frag_len has been negotiated or not. The second part is a test to see if the max_frag_len has been exceeded. In other words we are saying "if max_frag_len has been negotiated and it has been exceeded then error". Doing it the other way around says "if max_frag_len has been exceeded and it has been negotiated then error" - which seems somewhat nonsensical to me (if it hasn't been negotiated, then checking if it has been exceeded is meaningless).

Reply:
sorry, was conflating max_frag_len and ->length